lib/index.js

```diff
@@ -122,10 +122,15 @@ module.exports = {
       end: 'end',
       read: 'read',
     },
-    start: function (fn) {
-      process.emit('input', 'start')
+    start: function (...args) {
```

Comment:
```suggestion
    start: function (fn, ...args) {
```

I think baking in the `fn` is ok here.  We don't need to get so clever that we are allowing for anything else for the first param.

Reply:
In my diff you can also see the extra emit that includes `fn` if it wasn't a function.

Now that I compare the two I think either way works so let's not change this PR if we don't have to.  Your way is perfectly functional too!